doc/voq/voq_hld.md

```diff
@@ -0,0 +1,797 @@
+            "separator": ":",
+            "instance" : "redis"
+        },
+        ...
+        "CHASSIS_APP_DB" : {
+            "id" : 11,
+            "separator": ":",
+            "instance" : "redis-chassis_db"
+        },
+```
+        
+
+### 2.2.2 System Port Table
+A **new** table for system port configuration
+
+```
+SYSTEM_PORT:{{system_port_name}}
+    "system_port_id": {{index_number}}
+    "switch_id": {{index_number}}
+    "core_index": {{index_number}}
+    "core_port_index": {{index_number}}
+    "speed": {{index_number}}
+```
+### 2.2.3 ConfigDB Schemas
+**Existing** schema for DEVICE_METADATA in configuration DB
+```
+key                                   = DEVICE_METADATA|"localhost"      ; 
+; field                               = value
+switch_type                           = "voq" | "fabric"
+switch_id                             = 1*4DIGIT            ; number between 0 and 1023
+max_cores                             = 1*4DIGIT            ; max cores 1 and 1024
+    
+```
+
+```
+; Defines schema for VOQ System Port table attributes
+key                                   = SYSTEM_PORT:system_port_name ; VOQ system port name
+; field                               = value
+system_port_id                        = 1*5DIGIT                ; 1 to 32768
+switch_id                             = 1*4DIGIT                ; 0 to 1023 attached switch id
+core_index                            = 1*4DIGIT                ; 1 to 2048 switch core id
+core_port_index                       = 1*3DIGIT                ; 1 t0 256 port index in a core
+speed                                 = 1*7DIGIT                ; port line speed in Mbps
+```
+
+No changes in the schema of other CONFIG_DB tables. The name of the system ports used as key in the SYSTEM_PORT table is unique across chassis. The system_port_name can be same as the PORT table name as long as the PORT table name is unique across the chassis or it can be any character string which does not include ":". If a system port is used as inband interface, the name of that system port must be a name that will be accepted by kernel for netdevice name.
+
+For router interface and address configurations for the system ports, the existing INTERFACE table in CONFIG_DB is used.
+
+Please refer to the [schema](https://github.com/Azure/sonic-swss/blob/master/doc/swss-schema.md) document for details on value annotations. 
+
+## 2.3 CHASSIS_APP_DB
+This is a **new** database which resides in chassis redis server accessible by all sonic instances. This database is modeled as Application DB. The OrchAgent in all the sonic instances will write their local information such as INTERFACE, NEIGH, PORTCHANNEL and PORTCHANNEL_MEMBER tables to CHASSIS_APP_DB with hardware identifier's. The OrchAgent also reads sync-ed remote info such as SYSTEM_INTERFACE, SYSTEM_NEIGH, SYSTEM_PORTCHANNEL and SYSTEM_PORTCHANNEL_MEMBER tables of all sonic instances with hardware information such as hardware index.
+
```

Comment:
It's better to move this section "Chassis_app_db" up a bit, as the previous section also mentions "chassis_app_db".  Also can we make it clear whether chassis_app_db is only in control card, or will be in line cards as well?

Reply:
Srikanth, can you please do the edits necessary? To clarify, chassis_app_db will only run in the control card.